src/Cassandra/Metadata.cs

```diff
@@ -422,16 +422,84 @@ internal void ClearAggregate(string keyspaceName, string aggregateName, string[]
             }
         }
 
+        /// <summary>
+        /// Initiates a schema agreement check. This method does not perform retries so
+        /// <see cref="ProtocolOptions.MaxSchemaAgreementWaitSeconds"/> does not apply. 
+        /// </summary>
+        /// <returns>True if schema agreement was successful and false if it was not successful.</returns>
+        /// <exception cref="OperationCanceledException">If operation was canceled due to
+        /// the provided <see cref="CancellationToken"/></exception>
+        public async Task<bool> CheckSchemaAgreementAsync(CancellationToken ct = default(CancellationToken))
+        {
+            if (Hosts.Count == 1)
+            {
+                //If there is just one node, the schema is up to date in all nodes :)
+                return true;
+            }
+
+            try
+            {
+                var queries = new[]
+                {
+                    ControlConnection.QueryAsync(SelectSchemaVersionLocal).WithCancellation(ct),
+                    ControlConnection.QueryAsync(SelectSchemaVersionPeers).WithCancellation(ct)
+                };
+
+                await Task.WhenAll(queries).ConfigureAwait(false);
+
+                return CheckSchemaVersionResults(queries[0].Result, queries[1].Result);
+            }
+            catch (OperationCanceledException)
+            {
+                throw;
+            }
+            catch (Exception ex)
+            {
+                Logger.Error("Error while checking schema agreement.", ex);
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Checks if there is only one schema version between the provided query results.
+        /// </summary>
+        /// <param name="localVersionQuery">
+        /// Results obtained from a query to <code>system.local</code> table.
+        /// Must contain the <code>schema_version</code> column.
+        /// </param>
+        /// <param name="peerVersionsQuery">
+        /// Results obtained from a query to <code>system.peers</code> table.
+        /// Must contain the <code>schema_version</code> column.
+        /// </param>
+        /// <returns><code>True</code> if there is a schema agreement (only 1 schema version). <code>False</code> otherwise.</returns>
+        private bool CheckSchemaVersionResults(
+            IEnumerable<Row> localVersionQuery, IEnumerable<Row> peerVersionsQuery)
+        {
+            var versions = new HashSet<Guid>
```

Comment:
NIT: It could be simplified to use linq all the way: 

```C#
return new HashSet<Guid>(
    peerVersionsQuery
    .Concat(localVersionQuery)
    .Select(r => r.GetValue<Guid>("schema_version"))).Count == 1
```

Reply:
Nice! Changed it